aspnetcore/fundamentals/startup.md

```diff
@@ -38,35 +38,38 @@ In the following example from the default web site template, several extension m
+The [ConfigureServices](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.hosting.startupbase#Microsoft_AspNetCore_Hosting_StartupBase_ConfigureServices_Microsoft_Extensions_DependencyInjection_IServiceCollection_) method is optional; but if used, it's called before the `Configure` method by the web host. The web host may configure some services before ``Startup`` methods are called (see [hosting](xref:fundamentals/hosting). By convention, [Configuration options](xref:fundamentals/configuration) are set in this method.
 
 For features that require substantial setup there are `Add[Service]` extension methods on [IServiceCollection](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.dependencyinjection.iservicecollection). This example from the default web site template configures the app to use services for Entity Framework, Identity, and MVC:
 
 [!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
-Adding services to the services container makes them available within your application via [dependency injection](dependency-injection.md).
+Adding services to the services container makes them available within your application via [dependency injection](xref:fundamentals/dependency-injection).
 
 ## Services Available in Startup
 
-ASP.NET Core dependency injection provides application services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or one of its `Configure` or `ConfigureServices` methods. 
+ASP.NET Core dependency injection provides application services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
 
-Looking at each method in the `Startup` class in the order in which they are called, the following services may be requested as parameters:
+Below are some of the services typically requested by `Startup` methods:
 
-* In the constructor:  `IHostingEnvironment`, `ILoggerFactory`
-* In the `ConfigureServices` method:  `IServiceCollection`
-* In the `Configure` method:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`, `IApplicationLifetime`
+* In the constructor:  `IHostingEnvironment`, `ILogger<Startup>`
+* In `ConfigureServices`:  `IServiceCollection`
+* In `Configure`:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
```

Comment:
Because if you want to actually perform logging in Startup it's useful to DI a Logger<Startup> into it (and assign it to a field). However, all of the templates show configuring logging by passing a LoggerFactory into Configure and configuring it there (allowing it to be used in the app, but not in Startup). Again, in 2.0 this is fixed and logging is configured in WebHostBuilder, not Startup, but in 1.x this is what ships in the templates.